{"uid":"2dcd5472652c9d7d","name":"Exception test","fullName":"tests.forgot_password_test.TestForgotPassword#test_expected_exception_on_page_title","historyId":"288cd0030c92ba34b4ff4b58ed8ec476","time":{"start":1713662355931,"stop":1713662366193,"duration":10262},"description":"Exception catching","descriptionHtml":"<p>Exception catching</p>\n","status":"broken","statusMessage":"selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x5620bb294863 <unknown>\n#1 0x5620baf8a8c6 <unknown>\n#2 0x5620bafd5618 <unknown>\n#3 0x5620bafd56d1 <unknown>\n#4 0x5620bb018744 <unknown>\n#5 0x5620baff75cd <unknown>\n#6 0x5620bb015c19 <unknown>\n#7 0x5620baff7343 <unknown>\n#8 0x5620bafc8593 <unknown>\n#9 0x5620bafc8f5e <unknown>\n#10 0x5620bb25884b <unknown>\n#11 0x5620bb25c7a5 <unknown>\n#12 0x5620bb246571 <unknown>\n#13 0x5620bb25d332 <unknown>\n#14 0x5620bb22b87f <unknown>\n#15 0x5620bb283728 <unknown>\n#16 0x5620bb2838fb <unknown>\n#17 0x5620bb2939b4 <unknown>\n#18 0x7ff486294ac3 <unknown>","statusTrace":"self = <tests.forgot_password_test.TestForgotPassword object at 0x7fa7e13cd970>\n\n    @allure.description(\"Exception catching\")\n    @allure.title(\"Exception test\")\n    @allure.link(\"github.com/allure-examples/\", name=\"Allure Examples\")\n    @allure.issue(\"github.com/allure-examples/allure-examples/issues/1\", name=\"ISSUE-1\")\n    @allure.testcase(\n        \"github.com/allure-examples/allure-examples/issues/2\", name=\"TESTCASE-2\"\n    )\n    def test_expected_exception_on_page_title(self):\n>       self.about_page.click_login_link()\n\ntests/forgot_password_test.py:53: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \npages/about_page.py:20: in click_login_link\n    self.click(self.LOGIN_LINK)\npages/base_page.py:60: in click\n    el: WebElement = self.wait.until(\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"ca4ed027b09eee97e0b292af81cc62b5\")>\nmethod = <function element_to_be_clickable.<locals>._predicate at 0x7fa7e10e7ba0>\nmessage = ''\n\n    def until(self, method: Callable[[D], Union[Literal[False], T]], message: str = \"\") -> T:\n        \"\"\"Calls the method provided with the driver as an argument until the \\\n        return value does not evaluate to ``False``.\n    \n        :param method: callable(WebDriver)\n        :param message: optional message for :exc:`TimeoutException`\n        :returns: the result of the last call to `method`\n        :raises: :exc:`selenium.common.exceptions.TimeoutException` if timeout occurs\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.monotonic() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, \"screen\", None)\n                stacktrace = getattr(exc, \"stacktrace\", None)\n            time.sleep(self._poll)\n            if time.monotonic() > end_time:\n                break\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message: \nE       Stacktrace:\nE       #0 0x5620bb294863 <unknown>\nE       #1 0x5620baf8a8c6 <unknown>\nE       #2 0x5620bafd5618 <unknown>\nE       #3 0x5620bafd56d1 <unknown>\nE       #4 0x5620bb018744 <unknown>\nE       #5 0x5620baff75cd <unknown>\nE       #6 0x5620bb015c19 <unknown>\nE       #7 0x5620baff7343 <unknown>\nE       #8 0x5620bafc8593 <unknown>\nE       #9 0x5620bafc8f5e <unknown>\nE       #10 0x5620bb25884b <unknown>\nE       #11 0x5620bb25c7a5 <unknown>\nE       #12 0x5620bb246571 <unknown>\nE       #13 0x5620bb25d332 <unknown>\nE       #14 0x5620bb22b87f <unknown>\nE       #15 0x5620bb283728 <unknown>\nE       #16 0x5620bb2838fb <unknown>\nE       #17 0x5620bb2939b4 <unknown>\nE       #18 0x7ff486294ac3 <unknown>\n\n.venv/lib/python3.12/site-packages/selenium/webdriver/support/wait.py:105: TimeoutException","flaky":false,"newFailed":false,"newBroken":false,"newPassed":false,"retriesCount":0,"retriesStatusChange":false,"beforeStages":[{"name":"_verify_url","time":{"start":1713662160528,"stop":1713662160528,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"base_url","time":{"start":1713662160528,"stop":1713662160528,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"session_request","time":{"start":1713662160528,"stop":1713662160528,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false}],"testStage":{"description":"Exception catching","status":"broken","statusMessage":"selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x5620bb294863 <unknown>\n#1 0x5620baf8a8c6 <unknown>\n#2 0x5620bafd5618 <unknown>\n#3 0x5620bafd56d1 <unknown>\n#4 0x5620bb018744 <unknown>\n#5 0x5620baff75cd <unknown>\n#6 0x5620bb015c19 <unknown>\n#7 0x5620baff7343 <unknown>\n#8 0x5620bafc8593 <unknown>\n#9 0x5620bafc8f5e <unknown>\n#10 0x5620bb25884b <unknown>\n#11 0x5620bb25c7a5 <unknown>\n#12 0x5620bb246571 <unknown>\n#13 0x5620bb25d332 <unknown>\n#14 0x5620bb22b87f <unknown>\n#15 0x5620bb283728 <unknown>\n#16 0x5620bb2838fb <unknown>\n#17 0x5620bb2939b4 <unknown>\n#18 0x7ff486294ac3 <unknown>","statusTrace":"self = <tests.forgot_password_test.TestForgotPassword object at 0x7fa7e13cd970>\n\n    @allure.description(\"Exception catching\")\n    @allure.title(\"Exception test\")\n    @allure.link(\"github.com/allure-examples/\", name=\"Allure Examples\")\n    @allure.issue(\"github.com/allure-examples/allure-examples/issues/1\", name=\"ISSUE-1\")\n    @allure.testcase(\n        \"github.com/allure-examples/allure-examples/issues/2\", name=\"TESTCASE-2\"\n    )\n    def test_expected_exception_on_page_title(self):\n>       self.about_page.click_login_link()\n\ntests/forgot_password_test.py:53: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \npages/about_page.py:20: in click_login_link\n    self.click(self.LOGIN_LINK)\npages/base_page.py:60: in click\n    el: WebElement = self.wait.until(\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"ca4ed027b09eee97e0b292af81cc62b5\")>\nmethod = <function element_to_be_clickable.<locals>._predicate at 0x7fa7e10e7ba0>\nmessage = ''\n\n    def until(self, method: Callable[[D], Union[Literal[False], T]], message: str = \"\") -> T:\n        \"\"\"Calls the method provided with the driver as an argument until the \\\n        return value does not evaluate to ``False``.\n    \n        :param method: callable(WebDriver)\n        :param message: optional message for :exc:`TimeoutException`\n        :returns: the result of the last call to `method`\n        :raises: :exc:`selenium.common.exceptions.TimeoutException` if timeout occurs\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.monotonic() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, \"screen\", None)\n                stacktrace = getattr(exc, \"stacktrace\", None)\n            time.sleep(self._poll)\n            if time.monotonic() > end_time:\n                break\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message: \nE       Stacktrace:\nE       #0 0x5620bb294863 <unknown>\nE       #1 0x5620baf8a8c6 <unknown>\nE       #2 0x5620bafd5618 <unknown>\nE       #3 0x5620bafd56d1 <unknown>\nE       #4 0x5620bb018744 <unknown>\nE       #5 0x5620baff75cd <unknown>\nE       #6 0x5620bb015c19 <unknown>\nE       #7 0x5620baff7343 <unknown>\nE       #8 0x5620bafc8593 <unknown>\nE       #9 0x5620bafc8f5e <unknown>\nE       #10 0x5620bb25884b <unknown>\nE       #11 0x5620bb25c7a5 <unknown>\nE       #12 0x5620bb246571 <unknown>\nE       #13 0x5620bb25d332 <unknown>\nE       #14 0x5620bb22b87f <unknown>\nE       #15 0x5620bb283728 <unknown>\nE       #16 0x5620bb2838fb <unknown>\nE       #17 0x5620bb2939b4 <unknown>\nE       #18 0x7ff486294ac3 <unknown>\n\n.venv/lib/python3.12/site-packages/selenium/webdriver/support/wait.py:105: TimeoutException","steps":[{"name":"Click Login link","time":{"start":1713662355931,"stop":1713662366192,"duration":10261},"status":"broken","statusMessage":"selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x5620bb294863 <unknown>\n#1 0x5620baf8a8c6 <unknown>\n#2 0x5620bafd5618 <unknown>\n#3 0x5620bafd56d1 <unknown>\n#4 0x5620bb018744 <unknown>\n#5 0x5620baff75cd <unknown>\n#6 0x5620bb015c19 <unknown>\n#7 0x5620baff7343 <unknown>\n#8 0x5620bafc8593 <unknown>\n#9 0x5620bafc8f5e <unknown>\n#10 0x5620bb25884b <unknown>\n#11 0x5620bb25c7a5 <unknown>\n#12 0x5620bb246571 <unknown>\n#13 0x5620bb25d332 <unknown>\n#14 0x5620bb22b87f <unknown>\n#15 0x5620bb283728 <unknown>\n#16 0x5620bb2838fb <unknown>\n#17 0x5620bb2939b4 <unknown>\n#18 0x7ff486294ac3 <unknown>\n\n","statusTrace":"  File \"/home/runner/work/Selenium-Python-Example/Selenium-Python-Example/.venv/lib/python3.12/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/home/runner/work/Selenium-Python-Example/Selenium-Python-Example/pages/about_page.py\", line 20, in click_login_link\n    self.click(self.LOGIN_LINK)\n  File \"/home/runner/work/Selenium-Python-Example/Selenium-Python-Example/pages/base_page.py\", line 60, in click\n    el: WebElement = self.wait.until(\n                     ^^^^^^^^^^^^^^^^\n  File \"/home/runner/work/Selenium-Python-Example/Selenium-Python-Example/.venv/lib/python3.12/site-packages/selenium/webdriver/support/wait.py\", line 105, in until\n    raise TimeoutException(message, screen, stacktrace)\n","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":true,"stepsCount":0,"attachmentsCount":0,"hasContent":true,"attachmentStep":false},{"name":"public ip address","time":{"start":1713662366206,"stop":1713662366283,"duration":77},"status":"passed","steps":[],"attachments":[{"uid":"9d880f04d958c94f","name":"HTTP logs of http://checkip.amazonaws.com/","source":"9d880f04d958c94f.txt","type":"text/plain","size":359}],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":1,"hasContent":true,"attachmentStep":false}],"attachments":[{"uid":"2481c95b748ab8d1","name":"Cookies","source":"2481c95b748ab8d1.json","type":"application/json","size":3638},{"uid":"f7f7e2d1adb06cf","name":"Session Storage","source":"f7f7e2d1adb06cf.json","type":"application/json","size":359},{"uid":"38d7784a8443fbb0","name":"Local Storage","source":"38d7784a8443fbb0.json","type":"application/json","size":350},{"uid":"dac407a78855fd09","name":"Console Logs","source":"dac407a78855fd09.json","type":"application/json","size":7723},{"uid":"8e870bc0af31023","name":"Full Page Screenshot","source":"8e870bc0af31023.png","type":"image/png","size":6238629},{"uid":"4930023c26bbdaf5","name":"URL","source":"4930023c26bbdaf5.uri","type":"text/uri-list","size":23},{"uid":"89d4c36f4ec63f4","name":"Network Logs","source":"89d4c36f4ec63f4.json","type":"application/json","size":843004}],"parameters":[],"shouldDisplayMessage":true,"stepsCount":2,"attachmentsCount":8,"hasContent":true,"attachmentStep":false},"afterStages":[{"name":"session_request::0","time":{"start":1713662420990,"stop":1713662420991,"duration":1},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false}],"labels":[{"name":"story","value":"Forgot Password Feature's Functionality"},{"name":"severity","value":"critical"},{"name":"subSuite","value":"Custom sub suite"},{"name":"epic","value":"Security"},{"name":"suite","value":"Custom suite"},{"name":"parentSuite","value":"Custom parent suite"},{"name":"tag","value":"security"},{"name":"host","value":"fv-az1020-445"},{"name":"thread","value":"2409-MainThread"},{"name":"framework","value":"pytest"},{"name":"language","value":"cpython3"},{"name":"package","value":"tests.forgot_password_test"},{"name":"resultFormat","value":"allure2"}],"parameters":[],"links":[{"name":"TESTCASE-2","url":"https://github.com/allure-examples/allure-examples/issues/2","type":"tms"},{"name":"ISSUE-1","url":"https://github.com/allure-examples/allure-examples/issues/1","type":"issue"},{"name":"Allure Examples","url":"https://github.com/allure-examples/","type":"link"}],"hidden":false,"retry":false,"extra":{"severity":"critical","retries":[],"categories":[{"name":"Test defects","matchedStatuses":[],"flaky":false}],"tags":["security"]},"source":"2dcd5472652c9d7d.json","parameterValues":[]}